agent/src/main/java/reactor/blockhound/BlockHound.java

```diff
@@ -62,7 +62,13 @@
  */
 public class BlockHound {
 
-    static final String PREFIX = "$$BlockHound$$_";
+    /**
+     * The special methodName that should be used in e.g. {@link Builder#allowBlockingCallsInside(String, String)}
+     * when one wants to reference the static initializer block rather than a class method.
+     */
+    public static final String STATIC_INITIALIZER = "<clinit>";
```

Comment:
since this constant would never change (part of the JVM's spec), I do not feel that it makes sense to introduce it. We could keep the docs section (for both `<clinit>` and `<init>`), but the constant will only create an unnecessary noise in the public signature of this class.

Reply:
I don't know. Mentioning the reserved method name in the doc indeed feels like a minimum viable solution, but feels slightly less user-friendly than providing a safe constant which can quickly be auto-completed. Would it limit the noise drawback if the constant was defined in `Builder` instead of `BlockHound`? @royclarkson as a user, what do you think?

Reply:
I also prefer an explicit API with documentation as a minimum. That said, thinking more about the constant it also gets slightly verbose with the class name included, and it's not automatically discoverable. 

What about adding a sub-builder step:
```java
builder
    .allowBlockingCallsInside("com.example.ClassA", "methodA")
    .allowBlockingCallsInside("com.example.ClassB").staticInitializers()
    .disallowBlockingCallsInside("com.example.ClassC").constructors();
```

The sub-builder something like:
```java
interface BlockingCallSpec {

    BlockHound.Builder staticInitializers();

    BlockHound.Builder constructors();

}
```

Or perhaps an enum after all?
```java
builder
    .allowBlockingCallsInside("com.example.ClassA", "methodA")
    .allowBlockingCallsInside("com.example.ClassB", BlockHound.TargetType.STATIC_INITIALIZER)
    .disallowBlockingCallsInside("com.example.ClassC", BlockHound.TargetType.CONSTRUCTORS);
```

Reply:
exploring the builder idea in this PR and the enum in https://github.com/reactor/BlockHound/pull/184